packages/web-component/src/lib/helpers/helpers.ts

```diff
@@ -330,10 +375,21 @@ export const getChromiumVersion = (
 // As an optimization - We can show first screen if we have startScreenId and we don't have oidcIdpStateId
 // - If there startScreenId it means that the sdk can show the first screen and we don't need to wait for the sdk to return the first screen
 // - If there is a oidcIdpStateId - we can't skip this call because the sdk may
+// - If there is a samlIdpStateId - we can't skip this call because the sdk may
+// - If there is a samlIdpUsername - we can't skip this call because the sdk may
+// - If there is a ssoAppId - we can't skip this call because the sdk may
 export const showFirstScreenOnExecutionInit = (
   startScreenId: string,
-  oidcIdpStateId: string
-): boolean => startScreenId && !oidcIdpStateId;
+  oidcIdpStateId: string,
+  samlIdpStateId: string,
+  samlIdpUsername: string,
+  ssoAppId: string
```

Comment:
Do we want want to make the server call if we have one of these (`samlIdpUsername` || `ssoAppId`) and no `samlIdpStateId`

Reply:
its theoretic question/scenario as we will never have samlIdpUsername without have samlIdpStateId (or ssoAppId)..
Lets keep it simple and perform this call to the server for any change by any of those params (as we dont really have an extra calls here..)